src/main/java/com/googlecode/lanterna/terminal/swing/TerminalInputMethodRequests.java

```diff
@@ -0,0 +1,60 @@
+package com.googlecode.lanterna.terminal.swing;
+
+import com.googlecode.lanterna.TerminalPosition;
+
+import java.awt.*;
+import java.awt.font.TextHitInfo;
+import java.awt.im.InputMethodRequests;
+import java.text.AttributedCharacterIterator;
+
+class TerminalInputMethodRequests implements InputMethodRequests {
+
+    private Component owner;
+    private GraphicalTerminalImplementation terminalImplementation;
+    
+    public TerminalInputMethodRequests(Component owner, GraphicalTerminalImplementation terminalImplementation) {
+        this.owner = owner;
+        this.terminalImplementation = terminalImplementation;
+    }
+    
+    @Override
+    public Rectangle getTextLocation(TextHitInfo offset) {
+        Point location = owner.getLocationOnScreen();
+        TerminalPosition cursorPosition = terminalImplementation.getCursorPosition();
```

Comment:
This might be null if the cursor is hidden, in which case I think top-left corner is preferred

Reply:
Nevermind, I'm mixing things up. The position should always be valid.